sbin/functions.sh

```diff
@@ -964,6 +964,469 @@ clean_up()
+			swappartsize=`numfmt --from=auto $(cat $tmpfile)`
+			dialog --no-cancel --inputbox "Rootfs partition size:" 8 40 "2G" 2>$tmpfile
+			rootfspartsize=`numfmt --from=auto $(cat $tmpfile)`
+			dialog --no-cancel --inputbox "Containers partition size:" 8 40 \
+				"`numfmt --to=iec --round=down $(expr $size - ${bootpartsize} - ${swappartsize} - ${rootfspartsize})`" 2>$tmpfile
+			containerpartsize=`numfmt --from=auto $(cat $tmpfile)`
+		fi
+	fi
+	if [ "$install_type" == "installer" ]; then
+		echo "BOOTPART_START=\"0\"" >> ${localconf}
+		echo "BOOTPART_END=\"${bootpartsize}\"" >> ${localconf}
+		echo "BOOTPART_FSTYPE=\"fat32\"" >> ${localconf}
+		echo "BOOTPART_LABEL=\"OVERCBOOT\"" >> ${localconf}
+		echo "ROOTFS_START=\"${bootpartsize}\"" >> ${localconf}
+		if [ ${rootfspartsize} == "-1" ]; then
+			echo "ROOTFS_END=\"${rootfspartsize}\"" >> ${localconf}
+		else
+			echo "ROOTFS_END=\"$(expr `numfmt --from=auto $bootpartsize` + `numfmt --from=auto $rootfspartsize`)\"" >> ${localconf}
+		fi
+		echo "ROOTFS_FSTYPE=\"ext2\"" >> ${localconf}
+		echo "ROOTFS_LABEL=\"OVERCINSTROOTFS\"" >> ${localconf}
+	else
+		cat <<EOF > ${localpartitionlayout}
+d
+1
+d
+2
+d
+3
+d
+4
+
+
+n
+p
+1
+
++`numfmt --to=iec --from=iec ${bootpartsize}`
+n
+p
+2
+
++`numfmt --to=iec --from=iec ${swappartsize}`
+n
+p
+3
+
++`numfmt --to=iec --from=iec ${rootfspartsize}`
+n
+p
+
++`numfmt --to=iec --from=iec ${containerpartsize}`
+a
+1
+t
+1
+b
+t
+2
+82
+t
+3
+83
+w
+p
+q
+
+
+EOF
+		echo "FDISK_PARTITION_LAYOUT=\"${localpartitionlayout}\"" >> ${localconf}
+	fi
+}
+
+interactive_config()
+{
+	local device="$1"
+	local localconf_dir="$2"
+
+	if [ -z ${device} ]
+	then
+		debugmsg ${DEBUG_CRIT} "ERROR in interactive_config: device parameter not provided"
+		return 1
+	fi
+
+	if [ -z ${localconf_dir} ]
+	then
+		debugmsg ${DEBUG_CRIT} "ERROR in interactive_config: tmp config directory parameter not provided"
+		return 1
+	fi
+
+	local localconf="${localconf_dir}/config.sh"
+	tmpfile=$(mktemp)
+
+	echo "" > ${localconf}
+
+	# Distribution
+	dialog --no-cancel --inputbox "Distribution name:" 8 40 "OverC" 2>$tmpfile
+	local dialog_res=$(cat $tmpfile)
+	if [ -n "${dialog_res}" ]; then
+		echo "DISTRIBUTION=\"${dialog_res}\"" >> ${localconf}
+	else
+		echo "DISTRIBUTION=\"OverC\"" >> ${localconf}
+	fi
+
+	# Container selection
+	local filecount=0
+	local filelist="" menuitems=""
+	if [ -z "$ARTIFACTS_DIR" ]; then
+		if [ -e "tmp/deploy/images" ]; then
+			filecount=`ls tmp/deploy/images/ -1 | wc -l`
+			case "$filecount" in
+				0)
+					debugmsg ${DEBUG_CRIT}  "Project build incomplete. Please try finish the build first."
+					return 1
+					;;
+				1)
+					ARTIFACTS_DIR="`pwd`/tmp/deploy/images/`ls tmp/deploy/images/`"
+					;;
+				*)
+					filelist=`ls tmp/deploy/images` menuitems=""
+					for i in $filelist; do
+						menuitems="$menuitems $i $i "
+					done
+					dialog --no-cancel --menu "Found multiple machine target in your project, Please choose one:" \
```

Comment:
Or this. The installer knows nothing about machines, or what bitbake leaves as build outputs. Take a path to artefacts and use them.